scripts/clidocgen/main.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra/doc"
+
+	"github.com/coder/coder/cli"
+)
+
+type route struct {
+	Title       string  `json:"title,omitempty"`
+	Description string  `json:"description,omitempty"`
+	Path        string  `json:"path,omitempty"`
+	IconPath    string  `json:"icon_path,omitempty"`
+	State       string  `json:"state,omitempty"`
+	Children    []route `json:"children,omitempty"`
+}
+
+type manifest struct {
+	Versions []string `json:"versions,omitempty"`
+	Routes   []route  `json:"routes,omitempty"`
+}
+
+func main() {
+	// Set default configs for the docs
+	err := os.Setenv("CODER_CONFIG_DIR", "~/.config/coderv2")
+	if err != nil {
+		log.Fatal("Unable to set default value for CODER_CONFIG_DIR: ", err)
+	}
+
+	// Get the cmd CLI
+	cmd := cli.Root(cli.AGPL())
+
+	// Get paths
+	basePath := os.Getenv("BASE_PATH")
+	if basePath == "" {
+		log.Fatal("BASE_PATH should be defined")
+	}
+	markdownDocsDir := path.Join(basePath, "docs/cli")
+	manifestFilepath := path.Join(basePath, "docs/manifest.json")
+
+	// Generate markdown
+	err = doc.GenMarkdownTree(cmd, markdownDocsDir)
+	if err != nil {
+		log.Fatal("Error on generating CLI markdown docs: ", err)
+	}
+
+	// Create CLI routes
+	var cliRoutes []route
+	files, err := os.ReadDir(markdownDocsDir)
+	if err != nil {
+		log.Fatal("Error on loading docs/cli files: ", err)
+	}
+	for _, file := range files {
+		// Remove file extension and prefix
+		title := strings.Replace(file.Name(), ".md", "", 1)
+		title = strings.Replace(title, "coder_", "", 1)
+		title = strings.Replace(title, "_", " ", -1)
+
+		cliRoutes = append(cliRoutes, route{
+			Title: title,
+			Path:  "./cli/" + file.Name(),
+		})
+
+		// Remove non printable strings from generated markdown
+		// https://github.com/spf13/cobra/issues/1878
+		const ansi = "[\u001B\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;[a-zA-Z\\d]*)*)?\u0007)|(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PRZcf-ntqry=><~]))"
+		ansiRegex := regexp.MustCompile(ansi)
```

Comment:
nit: Oh, I meant, top top, for instance, line 29.

Reply:
Ahhh, I see. But I think would be better to have them together since they are only used here. It makes it easier to reason about IMO but no strong opinions.